test-frame-log-collector/src/main/java/io/skodjob/testframe/LogCollector.java

```diff
@@ -151,6 +171,22 @@ public void collectFromNamespaceToFolder(String namespaceName, String folderPath
         }
     }
 
+    /**
+     * Method that collects YAML of cluster wide resources
+     * {@link #rootFolderPath} with {@param folderPath}.
+     *
+     * @param folderPath        folder path for the log collection
+     */
+    public void collectClusterWideResourcesToFolder(String folderPath) {
+        String path = LogCollectorUtils.getFolderPath(rootFolderPath, folderPath);
+        createLogDirOnPath(path);
+        clusterWideResources.forEach(resource -> {
+            writeDataToFile(Paths.get(path, resource + ".yaml").toString(),
+                    kubeCmdClient.exec(false, false,
+                            "get", resource, "-o", "yaml").out());
+        });
```

Comment:
This will create one file per resource.
However I think that it will be much better to have each resource of the particular type in separate file - at least I found annoying checking a long file during some debug instead of opening one file where the particular resource is stored.

And in any case -> I would do it step by step to have it a bit clearer, so:

```
String resource = kubeCmd....;
String resFileName = LogCollectorUtils.getYamlFileNameForResource(resourceName);
String filePath = LogCollectorUtils.getFullPathForFolderPathAndFileName(path, resFileName);

writeDataToFile(filePath, resource);
```

it's much clearer IMHO

Reply:
If you would do it similarly with the folders as we have for the namespace resources, it can look like this:

```
/path/to/folder
|-my-namespace1
   |--secret
   |--deployment
|-my-namespace2
   |--secret
   |--deployment
|-cluster-wide-resources
   |--pv
       |---pv1.yaml
   |--node
       |---node1.yaml
```

WDYT? @Frawless @see-quick